### PR TITLE
Feature/viewvisitor+visitorout

### PR DESCRIPTION
**Pull request by GAOJIAXIN666**
add xmladapted visitor&visitorlist
add viewvisitor and viewvisitorpanel.fxml

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -328,40 +338,60 @@ Example(s):
...
 * `visitorin ic/S1234567A v/Amy`
 
 image::visitorin_screenshot.png[width=800]
+
+****
+WARNING: If the NRIC does not match that of any checked in patient in the system, an error message will be displayed.
+****
+
+****
+WARNING: User can't add visitor to a patient who has more than 5 visitors.
```
Comment (snajef) [inline docs/UserGuide.adoc]: Grammar mistake.

File: src/main/java/seedu/address/storage/XmlAdaptedVisitorList.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        this.visitorList = new ArrayList<>(vl);
+    }
+
+    @Override
+    public Iterator<XmlAdaptedVisitor> iterator() {
+        return visitorList.iterator();
+    }
+
+    @Override
+    //todo amend
```
Comment (omegafishy) [inline src/main/java/seedu/address/storage/XmlAdaptedVisitorList.java]: Has this todo been resolved ?

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -328,40 +338,60 @@ Example(s):
...
 * `visitorout ic/S1234567A v/Amy`
 
 image::visitorout_screenshot.png[width=800]
 
+****
+WARNING: If the NRIC does not match that of any checked in patient in the system, an error message will be displayed.
+****
+
+****
+WARNING: If the required visior in not in the patient's visitor list, an error message will be displayed.
```
Comment (snajef) [inline docs/UserGuide.adoc]: Spelling error.

File: src/main/java/seedu/address/model/visitor/VisitorList.java [hunk 4]
```diff
@@ -76,4 +80,16 @@ public boolean equals(Object o) {
 
         return false;
     }
+
+    /**
+     * Get a a copy of visitorlist for JavaFX TableView.
+     * @return A {@code ObservableList} of allergies.
```
Comment (snajef) [inline src/main/java/seedu/address/model/visitor/VisitorList.java]: Inaccurate comment.
Did you even vet through the code you copy-pasted?

File: src/main/java/seedu/address/storage/XmlAdaptedVisitor.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * JAXB-friendly adapted version of the Visitor class.
+ * @author GAO JIAXIN666
+ */
+@XmlRootElement
+public class XmlAdaptedVisitor {
+    @XmlElement(required = true)
+    private String visitor;
+
+    /**
+     * Constructs an XmlAdaptedDiagnosis.
```
Comment (snajef) [inline src/main/java/seedu/address/storage/XmlAdaptedVisitor.java]: Clearly inaccurate comment.
Please (at the very least) edit your comments to be accurate if you're going to copy-paste code.

File: src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    @Test
+    public void equals_originalAndFromModelTypeCopy_returnsTrue() {
+        XmlAdaptedVisitor xv = new XmlAdaptedVisitor(visitor);
+        XmlAdaptedVisitor xvCopy = new XmlAdaptedVisitor (xv.toModelType());
+
+        assertTrue(xv.equals(xvCopy));
+    }
+
+    @Test
+    public void equals_differntVisitorName_returnsFalse() {
```
Comment (omegafishy) [inline src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java]: Typo here.

File: src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.visitor.Visitor;
+
+public class XmlAdaptedVisitorTest {
+    public static final String VALID_VISITOR = "GAO JIAXIN";
+    public static final String VALID_VISITOR1 = "AMY";
+
+    private Visitor visitor;
+    private Visitor visitor1;
```
Comment (omegafishy) [inline src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java]: Naming convention can be improved, use a variable name like maybe anotherVisitor or differentVisitor. A name that explains what the variable is.

File: src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Before
+    public void setUp() throws IllegalValueException {
+        visitor = new Visitor(VALID_VISITOR);
+        visitor1 = new Visitor(VALID_VISITOR1);
```
Comment (omegafishy) [inline src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java]: Same here

File: src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    @Test
+    public void toModelType_validPersonDetails_returnsPerson() {
+        XmlAdaptedVisitor xmlAdaptedVisitor = new XmlAdaptedVisitor(visitor);
+        assertEquals(visitor, xmlAdaptedVisitor.toModelType());
+    }
+
+    @Test
+    public void equals_copyOfSource_returnsTrue() {
+        XmlAdaptedVisitor xv = new XmlAdaptedVisitor(VALID_VISITOR);
+        XmlAdaptedVisitor xv1 = new XmlAdaptedVisitor(visitor);
```
Comment (omegafishy) [inline src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java]: Same here.

**Review by darienchong — CHANGES_REQUESTED** [review]
Multiple grammar and spelling errors throughout.
Blatantly inaccurate comments on classes.
The only saving grace is that the code was largely copy-pasted from other portions of the codebase, so there are no issues there.

Additionally, this PR contains changes for your PPP AND your commands.
Please separate your PR contents into their appropriate branches before opening their respective PRs.

Overall, very dirty PR with low-quality changes.

File: src/main/java/seedu/address/storage/XmlAdaptedVisitorList.java (hunk 2 above)
Comment (GAOJIAXIN666) [reply src/main/java/seedu/address/storage/XmlAdaptedVisitorList.java]: forget to remove it, sry

File: src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java (hunk 7 above)
Comment (GAOJIAXIN666) [reply src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java]: Thx. Renamed

File: src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java (hunk 8 above)
Comment (GAOJIAXIN666) [reply src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java]: Thx. Renamed

File: src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java (hunk 9 above)
Comment (GAOJIAXIN666) [reply src/test/java/seedu/address/storage/XmlAdaptedVisitorTest.java]: Thx. Renamed

**Review by lingzhiyu — CHANGES_REQUESTED** [review]
Correct the language mistakes pointed out, and correct the method documentations.

**Review by lingzhiyu — APPROVED** [review]

**Pull request merged**